src/lib/FiatApi.js

```diff
@@ -11,109 +11,138 @@ class FiatApi {
      * >>>}
      */
     static async getExchangeRates(cryptoCurrencies, vsCurrencies) {
-        const coinIds = cryptoCurrencies.map(currency => FiatApi.COINGECKO_COIN_IDS[currency]);
-        const response = await fetch(
-            `${FiatApi.API_URL}/simple/price?ids=${coinIds.join(',')}&vs_currencies=${vsCurrencies.join(',')}`,
-        );
-        if (!response.ok) throw new Error(`Failed to fetch exchange rates: ${response.status}`);
-        const apiResult = await response.json();
-        // Map coingecko coin ids back to SupportedCryptoCurrency enum and sanitize retrieved data.
-        return cryptoCurrencies.reduce((result, cryptoCurrency) => {
-            const record = apiResult[FiatApi.COINGECKO_COIN_IDS[cryptoCurrency]];
-            const sanitizedRecord = Object.keys(record).reduce((sanitized, fiatCurrency) => ({
-                ...sanitized,
-                ...(Object.values(FiatApi.SupportedFiatCurrency).includes(/** @type {any} */ (fiatCurrency))
-                    ? { [fiatCurrency]: parseFloat(record[fiatCurrency]) }
-                    : null
-                ),
-            }), {});
-            return {
-                ...result,
-                [cryptoCurrency]: sanitizedRecord,
-            };
-        }, {});
+        switch (FiatApi.Provider) {
+            case FiatApi.SupportedProvider.CoinGecko: {
+                const coinIds = cryptoCurrencies.map(currency => FiatApi.COINGECKO_COIN_IDS[currency]);
+                const response = await fetch(
+                    `${FiatApi.API_URL}/simple/price?ids=${coinIds.join(',')}&vs_currencies=${vsCurrencies.join(',')}`,
+                );
+                if (!response.ok) throw new Error(`Failed to fetch exchange rates: ${response.status}`);
+                const apiResult = await response.json();
+                // Map coingecko coin ids back to SupportedCryptoCurrency enum and sanitize retrieved data.
+                return cryptoCurrencies.reduce((result, cryptoCurrency) => {
+                    const record = apiResult[FiatApi.COINGECKO_COIN_IDS[cryptoCurrency]];
+                    const sanitizedRecord = Object.keys(record).reduce((sanitized, fiatCurrency) => ({
+                        ...sanitized,
+                        ...(Object.values(FiatApi.SupportedFiatCurrency).includes(/** @type {any} */ (fiatCurrency))
+                            ? { [fiatCurrency]: parseFloat(record[fiatCurrency]) }
+                            : null
+                        ),
+                    }), {});
+                    return {
+                        ...result,
+                        [cryptoCurrency]: sanitizedRecord,
+                    };
+                }, {});
+            }
+
+            default:
+                throw new Error(`FiatApi.Provider ${FiatApi.Provider} not supported yet.`);
```

Comment:
This `default` should not be necessary. I think it might even be detrimental, as with it there won't be a TS compiler error when we add a new Provider, it'll only error at runtime.
My suggestion is to remove it.

Reply:
Our TS version does in fact not warn if a provider is missing in the switch statement, but as a newer version might, I changed it.

Reply:
See https://github.com/nimiq/keyguard/pull/467, which you are welcome to review ;)